crates/red_knot_python_semantic/src/module_resolver/path.rs

```diff
@@ -59,6 +59,15 @@ impl ModulePath {
         self.relative_path.push(component);
     }
 
+    #[must_use]
+    pub(super) fn is_file_module(&self, resolver: &ResolverContext) -> bool {
+        self.with_pyi_extension().to_file(resolver).is_some()
+            || self
+                .with_py_extension()
+                .and_then(|path| path.to_file(resolver))
+                .is_some()
+    }
```

Comment:
I think I would call this method something like `with_any_python_extension()`, and have it return `Option<File>`. Callers of the method from `resolver.rs` that just need to know whether the path exists on disk with a Python extension can then just do `path.with_any_python_extensions(resolver).is_some()`, and we'll be able to remove the `resolve_file_module()` function from `resolver.rs` (because this will then do the same thing as that function):

```suggestion
    /// If `self` exists on disk with either a `.pyi` or `.py` extension,
    /// return the [`File`] corresponding to that path.
    ///
    /// `.pyi` files take priority, as they always have priority when
    /// resolving modules.
    #[must_use]
    pub(super) fn with_any_python_extension(&self, resolver: &ResolverContext) -> Option<File> {
        self.with_pyi_extension().to_file(resolver).or_else(|| {
            self
                .with_py_extension()
                .and_then(|path| path.to_file(resolver))
            })
    }
```

Reply:
oh, I think you addressed this moments before I commented :-)

Reply:
So I think the method exists, right now it's named `is_file_module`, but I can agree on any name :)
It's also been moved to `resolve.rs` under MichaReiser's comments that it didn't really match existing methods in `ModulePath`

Reply:
Yup, I agree that it doesn't really belong in `path.rs`, `resolver.rs` is better!

Reply:
I'll add the docstring though, I think it helps :)